src/main/java/com/example/pokebattlez/battle/controller/Battle.java

```diff
@@ -0,0 +1,72 @@
+package com.example.pokebattlez.battle.controller;
+
+import com.example.pokebattlez.battle.model.Pokemon;
+import com.example.pokebattlez.battle.model.Stat;
+import com.example.pokebattlez.battle.service.BattleService;
+import lombok.Builder;
+import lombok.Data;
+
+import java.util.*;
+
+public class Battle {
+    private UUID id;
+
+    private final List<Long> trainers = new ArrayList<>();
+    private final Map<Long, List<Pokemon>> pokemon = new HashMap<>();
+    private final Map<Long, Pokemon> activePokemon = new HashMap<>();
+    private final List<PlayerAction> playerActions = new ArrayList<>();
```

Comment:
Overall a Battle consist of some amount of trainers with a list of Pokemons mapped to each, out of which one is only active.  
Does order of the Pokemon in the list of trainer's Pokemons matter?

Great that you thought of using Maps. Maybe this could be simplified to a map of <Trainer, List<Pokemon>> and the active pokemon could be singled out by an attribute (this way there's no need for a separate list).
In general if the objects are somehow related it is good to keep them in the common data structure, so that they are syncronzied.

Reply:
Each pokemon has a position attribute that gets used for order-specific operations so the order in the list doesn't matter.

Each trainer has at most 6 pokemon, with only one pokemon active per trainer. So the pokemon in activePokemon are actually just references to the ones in the pokemon map. I choose this approach since I need access to the active pokemon pretty often so filtering each time would be ineffective.